pandas/_libs/algos.pyx

```diff
@@ -1001,6 +1001,7 @@ def rank_1d(
         ndarray[uint8_t, ndim=1] mask
         bint keep_na, at_end, next_val_diff, check_labels, group_changed
         rank_t nan_fill_val
+        float64_t grp_size
```

Comment:
Should this not be an integer?

Reply:
Did this because `grp_sizes` was defined as an array of `float64_t` (I'd guess to avoid worrying about casting to float when dividing by `grp_sizes` at the end). Can change both to int if you think that would make more sense.

Reply:
Ah makes sense. That feels like a mistake so yea if you can change the array and nothing breaks lets do that. If it causes an issue can do in a follow up

Reply:
Have changed in latest commit